cache.go

```diff
@@ -205,29 +208,35 @@ func NewCache(config *Config) (*Cache, error) {
 	//       goroutines we have running cache.processItems(), so 1 should
 	//       usually be sufficient
 	go cache.processItems()
+
 	return cache, nil
 }
 
-func (c *Cache) Wait() {
+func (c *Cache[K, V]) Wait() {
 	if c == nil || c.isClosed {
 		return
 	}
 	wg := &sync.WaitGroup{}
 	wg.Add(1)
-	c.setBuf <- &Item{wg: wg}
+	c.setBuf <- &Item[V]{wg: wg}
 	wg.Wait()
 }
 
 // Get returns the value (if any) and a boolean representing whether the
 // value was found or not. The value can be nil and the boolean can be true at
 // the same time.
-func (c *Cache) Get(key interface{}) (interface{}, bool) {
-	if c == nil || c.isClosed || key == nil {
-		return nil, false
+func (c *Cache[K, V]) Get(key K) (V, bool) {
+	if c == nil || c.isClosed {
+		return zeroValue[V](), false
 	}
 	keyHash, conflictHash := c.keyToHash(key)
+
+	// If key in a pointer type and nil the default keyToHash function will return 0,0
+	if keyHash == 0 && conflictHash == 0 {
+		return zeroValue[V](), false
+	}
```

Comment:
i agree that &key == nil is redundant but a pointer to K (*K) as the arg is not the right way to go imo.

@mangalaman93 if key is nil then 
`keyHash, conflictHash := c.keyToHash(key)`
i have added that will return 0,0 in the default keyToHash, if users want to implement their own keyToHash and also use a pointer type as their key they should implement the function to return 0,0 in case of nil. WDYT?